src/main/kotlin/com/mineinabyss/launchy/logic/LaunchyState.kt

```diff
@@ -117,6 +134,13 @@ class LaunchyState(
             downloading -= mod
             downloadURLs[mod] = mod.url
             save()
+
+            if (mod.configUrl != null && configToggleState) {
+                Downloader.download(url = mod.configUrl, writeTo = Dirs.configZip)
```

Comment:
Use `launch` to launch this and the other download at once on `Dispatchers.IO`, then join both jobs. Scyu can help you with this, as well there are pretty detailed docs for kotlin coroutines, I'd recommend going through them slowly because there's some good stuff to know!

Reply:
Leaving as is for now since the mod download failing should prevent config download and the configs are so tiny anyways the speedup is negligible.